Tasks/AzureKeyVault/models/KeyVaultTaskParameters.ts

```diff
@@ -0,0 +1,45 @@
+/// <reference path="../../../definitions/node.d.ts" /> 
+/// <reference path="../../../definitions/vsts-task-lib.d.ts" /> 
+
+import tl = require("vsts-task-lib/task");
+import msRestAzure = require("./../operations/azure-rest/azure-arm-common");
+import util = require("util")
+
+export class KeyVaultTaskParameters {
+
+    public subscriptionId: string;
+    public keyVaultName: string;
+    public secretsFilter: string[];
+    public vaultCredentials: msRestAzure.ApplicationTokenCredentials;
+    public keyVaultUrl: string;
+
+    constructor() {
+        try {
+            var connectedService = tl.getInput("ConnectedServiceName", true);
+            this.subscriptionId = tl.getEndpointDataParameter(connectedService, "SubscriptionId", true);
+            this.keyVaultName = tl.getInput("KeyVaultName", true);
+            this.secretsFilter = tl.getDelimitedInput("SecretsFilter", ",", true);
+
+            var azureKeyVaultDnsSuffix = tl.getEndpointDataParameter(connectedService, "AzureKeyVaultDnsSuffix", true);
+
+            this.keyVaultUrl = util.format("https://%s.%s", this.keyVaultName, azureKeyVaultDnsSuffix);
+            this.vaultCredentials = this.getVaultCredentials(connectedService, azureKeyVaultDnsSuffix);
+        }
+        catch (error) {
```

Comment:
you shouldn't catch in a contructor.  The consumer can catch this.  You're just masking what the root issue is.